src/main/java/com/at/ac/tuwien/sepm/ss15/edulium/dao/impl/MenuCategoryDAOImpl.java

```
@@ -0,0 +1,206 @@
+package com.at.ac.tuwien.sepm.ss15.edulium.dao.impl;
+
+import com.at.ac.tuwien.sepm.ss15.edulium.dao.DAOException;
+import com.at.ac.tuwien.sepm.ss15.edulium.dao.MenuCategoryDAO;
+import com.at.ac.tuwien.sepm.ss15.edulium.domain.MenuCategory;
+import com.at.ac.tuwien.sepm.ss15.edulium.domain.validation.MenuCategoryValidator;
+import com.at.ac.tuwien.sepm.ss15.edulium.domain.validation.ValidationException;
+import org.springframework.beans.factory.annotation.Autowired;
+
+import javax.sql.DataSource;
+import java.sql.*;
+import java.util.ArrayList;
+import java.util.List;
+
+/**
+ * H2 Database Implementation of the MenuCategoryDAO interface
+ */
+class MenuCategoryDAOImpl implements MenuCategoryDAO {
+    @Autowired
+    private DataSource dataSource;
+    @Autowired
+    private MenuCategoryValidator validator;
+
+    /**
+     * writes the object into the database and sets the identity parameter of
+     * menuCategory
+     * @param menuCategory object to store
+     * @throws DAOException if an error accessing the database occurred
+     */
+    @Override
+    public void create(MenuCategory menuCategory) throws DAOException, ValidationException {
+        assert(menuCategory != null);
```

menuCategory == null should throw a validation exception instead (move the is null check to menu category validator)

same applies for all other assert(menuCategory != null) in this class